app/locations/controllers.js

```diff
@@ -0,0 +1,28 @@
+const { find, groupBy, sortBy } = require('lodash')
+
+function _getUserLocations(user) {
+  return groupBy(sortBy(user.locations, 'title'), 'location_type')
+}
+
+function locations(req, res) {
+  const userLocations = _getUserLocations(req.session.user)
+  res.render('locations/views/locations.njk', { userLocations })
+}
+
+function setLocation(req, res) {
+  const locationId = req.params.locationId
+  const location = find(req.session.user.locations, ['id', locationId])
+
+  if (locationId && !location) {
+    throw new Error(`Not authorised to switch to location ${locationId}`)
+  } else if (location) {
+    req.session.currentLocation = location
+  }
```

Comment:
To make it a little clearer rather than an `if, else if` you could do:

```
if (locationId && !location) {
    throw new Error(`Not authorised to switch to location ${locationId}`)
}

if (location) {
    req.session.currentLocation = location
}
```